scripts/modules/b2b-account/account-hierarchy.js

```diff
@@ -59,8 +61,44 @@ define(["modules/jquery-mozu", 'modules/api', "underscore", "hyprlive", "modules
             var accountId = e.currentTarget.dataset.mzValue;
         },
         changeParentAccount: function (e) {
-            //todo: need to implement this method in future.
             var accountId = e.currentTarget.dataset.mzValue;
+            var self = this;
+            inValidParentAccounts = [];
+            var validParentAccounts = [];
+            var currentAccount = this.getAccount(accountId, userChildAccounts);
+            var parentAccount = this.getAccount(currentAccount.account.parentAccountId, self.model.apiModel.data.accounts);
```

Comment:
I don't like how `userChildAccounts` is an array of objects with an account property. If you can refactor this to be a simple list of accounts or account IDs that would be better/cleaner. I had asked Hrishikesh to implement this as a simple list of accounts before.

The name `userChildAccounts` is also misleading. The name `supportedParentAccounts` or maybe `associatedAccountDescendants` is more accurate. I much prefer the first since it's an array used purely for the parent account list.

Reply:
Changed variable name to **supportedParentAccounts** . I kept it as is an array of objects as I don't know the impact on hrishi's story.